config/scout_apm.yml

```diff
@@ -0,0 +1,49 @@
+# This configuration file is used for Scout APM.
+# Environment variables can also be used to configure Scout. See our help docs at https://scoutapm.com/docs/ruby/configuration#environment-variables for more information.
+common: &defaults
+
+  # key: Your Organization key for Scout APM. Found on the settings screen.
+  # - Default: none
+  # key: We set this ENV `SCOUT_KEY`
+
+  # log_level: Verboseness of logs.
+  # - Default: 'info'
+  # - Valid Options: debug, info, warn, error
+  # log_level: debug
+  # TIP: this can be set via ENV `SCOUT_LOG_LEVEL` to change without redeploy
+
+  # use_prepend: Use the newer `prepend` instrumentation method. In some cases, gems
+  #              that use `alias_method` can conflict with gems that use `prepend`.
+  #              To avoid the conflict, change this setting to match the method
+  #              that the other gems use.
+  #              If you have another APM gem installed, such as DataDog or NewRelic,
+  #              you will likely want to set `use_prepend` to true.
+  #
+  #              See https://scoutapm.com/docs/ruby/configuration#library-instrumentation-method
+  #              for more information.
+  # - Default: false
+  # - Valid Options: true, false
+  # use_prepend: true
+
+  # name: Application name in APM Web UI
+  # - Default: the application names comes from the Rails or Sinatra class name
+  # name:
+
+  # monitor: Enable Scout APM or not
+  # - Default: none
+  # - Valid Options: true, false
+  monitor: true
+
+production:
+  <<: *defaults
+
+development:
+  <<: *defaults
+  monitor: false
+
+test:
+  <<: *defaults
+  monitor: false
+
+staging:
```

Comment:
Non-blocking, but I'm curious about what the definition of "staging" is in Scout terms. I see above that we're blocking test and dev tiers from sending anything to Scout, but we're also not blocking staging - so I'm not sure what Scout's worldview is for the various Heroku tiers.

Reply:
This is a generated config file. We don't name any Environments `staging` so it is never used. All of our Heroku Environments are `production` as far as Rails is concerned. We'll very likely just explicitly disable (via `SCOUT_MONITOR: false` in PR builds. I'm not sure we'll know exactly what we'll need until we are a bit further into this... at which point I hope we remember to go back and update the docs with our actual implementation...